### PR TITLE
Feature oop

### PR DESCRIPTION
**Pull request by peterdgreat**
## Created the remaining classes for the school library and created the associations between them.

### CODE REVIEW

File: Classes/rental.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Rental
+  attr_accesor :date, :book, :person
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
+end
```
Comment (DeliceLydia) [inline Classes/rental.rb]: - [ ] # *The rental should be added itself to the @book achieving the 'belongs-to' association
``book.rentals << self``
- [ ] # *The rental is adding itself to the @person the 'belongs-to' association
`person.rentals << self`

File: Classes/person.rb [hunk 2]
```diff
@@ -30,4 +31,8 @@ def can_use_services?
   def validate_name
     @corrector.correct_name
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (DeliceLydia) [inline Classes/person.rb]: - [ ] `The person` itself should be sent as a parameter to create the new rental achieving the 'has-many' association. New rental should be created with `date`, `book`, and `person` itself

File: Classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (DeliceLydia) [inline Classes/book.rb]: - [ ] `*the book` itself should be sent as a parameter to create the new rental achieving the 'has-many' association. a new rental should be created with `date`, `person` and `book` itselff

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @peterdgreat, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Classes/person.rb (hunk 2 above)
Comment (peterdgreat) [reply Classes/person.rb]: Do you mean writing  :
rentals.person = self

File: Classes/book.rb (hunk 3 above)
Comment (peterdgreat) [reply Classes/book.rb]: Do you mean writing  
rentals.book =self

File: Classes/rental.rb (hunk 1 above)
Comment (peterdgreat) [reply Classes/rental.rb]: Thanks Man

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @peterdgreat , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Pull request merged**